j5/boards/base.py

```diff
@@ -80,18 +77,20 @@ def __init__(self, board: Board, backend: Backend):
         self.board_class: Board = board
         self._backend: Backend = backend
         self._iterator_counter: int = 0
-        self.boards: List[Board] = []
+        self.boards: Dict[str, Board] = {}
 
         self.update_boards()
 
     def update_boards(self) -> None:
         """Update the boards in this group to see if new boards have been added."""
-        self.boards = self.board_class.discover(self._backend)
+        self.boards = {}  # type: Dict[str, Board]
```

Comment:
Can we use post Python 3.5 styles typing hints here please. I'm not entirely sure if we even need to redefine the type here as it is already defined in the constructor

Reply:
We could declare it at the beginning of the class before the constructor so that it would be more visible in the future. Not sure if this would be the 'python way' of doing things tho'.
Opinion?

Reply:
Try `self.boards: Dict[str,Board] = {}`

Reply:
I'm still not sure if type hinting is even necessary there since we already declared self.boards in the constructor which is just a few lines of code above the update_boards method. I will include the change now but I will like your feedback on this matter.